lib/bolt/result.rb

```diff
@@ -7,47 +7,61 @@ module Bolt
   class Result
     attr_reader :target, :value, :action, :object
 
-    def self.from_exception(target, exception, action: 'action')
+    def self.from_exception(target, exception, action: 'action', position: [])
+      details = create_details(position)
       if exception.is_a?(Bolt::Error)
-        error = exception.to_h
+        error = { 'details' => details }.merge(exception.to_h)
       else
+        details['class'] = exception.class.to_s
         error = {
           'kind' => 'puppetlabs.tasks/exception-error',
           'issue_code' => 'EXCEPTION',
           'msg' => exception.message,
-          'details' => { 'class' => exception.class.to_s }
+          'details' => details
         }
         error['details']['stack_trace'] = exception.backtrace.join('\n') if exception.backtrace
       end
       Result.new(target, error: error, action: action)
     end
 
-    def self.for_command(target, stdout, stderr, exit_code, action, command)
+    def self.create_details(position)
+      details = {}
+      details['file'] = position[0] if position.any?
+      details['line'] = position[1] if position[1]
+      details
```

Comment:
```suggestion
      details = {}
      details['file'] = position[0] if position[0]
      details['line'] = position[1] if position[1]
      details
```

It looks like the stack will either always have [both file and line](https://github.com/puppetlabs/puppet/blob/eadd5474c26e7d28d90de00b9d7a7545ac10e55d/lib/puppet/pops/puppet_stack.rb#L33-L34) or return an [empty array](https://github.com/puppetlabs/puppet/blob/eadd5474c26e7d28d90de00b9d7a7545ac10e55d/lib/puppet/pops/puppet_stack.rb#L57), so I'm not sure there needs to be a check that both values exist.

Could even do a fun little one-liner: `%w[file line].zip(position).to_h.compact` :P